<a href="https://colab.research.google.com/github/manumc2306-sudo/Lenguaje-de-programacion/blob/main/The_Trading_Floor_Experience_Criptomonedas.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Presentación del Fondo de Inversión en Criptomonedas: Optimizando Capital con Datos Históricos

**¡Buenas tardes, distinguidos inversionistas!**

Somos un equipo dedicado a la gestión de capital en el dinámico mercado de las criptomonedas. Nuestra misión es clara: construir una aplicación interactiva que analice datos históricos para proponer y defender una estrategia de trading que optimice su capital. Hoy, estamos aquí para presentarles la mejor opción para sus inversiones.

**INSTALACION DE LIBRERIA**

In [101]:
# Celda 1: instalar y importar
!pip install yfinance --quiet

print("¡Listo para analizar criptos y decidas que somos tu mejor opcion!")
print("Las librerias las iremos importando a lo largo del analisis para que entiendas cada paso y entiendas nuestro fondo de mejor manera.")

¡Listo para analizar criptos y decidas que somos tu mejor opcion!
Las librerias las iremos importando a lo largo del analisis para que entiendas cada paso y entiendas nuestro fondo de mejor manera.


**DESCARGA DE DATOS**

### Nuestro Enfoque Basado en Datos: Adquisición y Preparación

Para garantizar decisiones de inversión informadas, hemos implementado una robusta metodología de análisis que comienza con la adquisición de datos históricos. Hemos descargado series temporales **diarias** de las criptomonedas más relevantes y con mayor capitalización de mercado para el año **2025** (desde el 1 de enero hasta la fecha actual):

*   **Bitcoin (BTC):** El referente del mercado cripto, un activo fundamental en cualquier cartera digital.
*   **Ethereum (ETH):** La plataforma líder para aplicaciones descentralizadas y contratos inteligentes, con un ecosistema en constante expansión.
*   **Solana (SOL):** Una blockchain de alto rendimiento, reconocida por su velocidad y escalabilidad, que representa una apuesta por la innovación.
*   **Cardano (ADA):** Una plataforma blockchain con un enfoque en la sostenibilidad y la escalabilidad a través de un enfoque científico.

Estos datos detallados, que incluyen precios de apertura, cierre, máximos, mínimos y volumen, nos proporcionan una visión completa para evaluar el comportamiento pasado y proyectar escenarios futuros.

In [129]:
import yfinance as yf

btc = yf.download('BTC-USD',
  start='2025-01-01',
  end='2026-05-26',
  interval='1d')

eth = yf.download('ETH-USD',
  start='2025-01-01',
  end='2026-05-26',
  interval='1d')

sol = yf.download('SOL-USD',
  start='2025-01-01',
  end='2026-05-26',
  interval='1d')

ada = yf.download('ADA-USD',
  start='2025-01-01',
  end='2026-05-26',
  interval='1d')

/tmp/ipykernel_33471/2785521071.py:3: FutureWarning:

YF.download() has changed argument auto_adjust default to True

[*********************100%***********************]  1 of 1 completed
/tmp/ipykernel_33471/2785521071.py:8: FutureWarning:

YF.download() has changed argument auto_adjust default to True

[*********************100%***********************]  1 of 1 completed
/tmp/ipykernel_33471/2785521071.py:13: FutureWarning:

YF.download() has changed argument auto_adjust default to True

[*********************100%***********************]  1 of 1 completed
/tmp/ipykernel_33471/2785521071.py:18: FutureWarning:

YF.download() has changed argument auto_adjust default to True

[*********************100%***********************]  1 of 1 completed


**DATAFRAMES** (Para validar la conexion)

In [130]:
btc.head()

Price,Close,High,Low,Open,Volume
Ticker,BTC-USD,BTC-USD,BTC-USD,BTC-USD,BTC-USD
Date,,,,,
2025-01-01,94419.757812,94929.867188,92788.125000,93425.101562,24519888919
2025-01-02,96886.875000,97739.820312,94201.570312,94416.289062,46009564411
2025-01-03,98107.429688,98956.914062,96034.617188,96881.726562,35611391163
2025-01-04,98236.226562,98734.429688,97562.976562,98106.992188,22342608078
2025-01-05,98314.960938,98813.304688,97291.765625,98233.906250,20525254825


In [131]:
eth.head()

Price,Close,High,Low,Open,Volume
Ticker,ETH-USD,ETH-USD,ETH-USD,ETH-USD,ETH-USD
Date,,,,,
2025-01-01,3353.504150,3366.531494,3310.255859,3332.406494,14195410493
2025-01-02,3451.392578,3493.448242,3348.352051,3353.412109,22243574698
2025-01-03,3605.009766,3627.055664,3421.831543,3451.680908,21877299255
2025-01-04,3657.706787,3669.196777,3574.326904,3605.200684,16060610759
2025-01-05,3634.103760,3673.812988,3594.616943,3657.743164,12830306908


In [132]:
sol.head()

Price,Close,High,Low,Open,Volume
Ticker,SOL-USD,SOL-USD,SOL-USD,SOL-USD,SOL-USD
Date,,,,,
2025-01-01,193.873734,194.818497,187.879456,189.266922,2324231668
2025-01-02,207.767044,209.066727,193.258530,193.872131,4230329837
2025-01-03,217.747330,219.559525,205.652405,207.767609,4064420852
2025-01-04,216.622925,217.905045,213.831451,217.748047,2432220391
2025-01-05,213.393631,218.878403,211.765228,216.622757,2403765299


In [133]:
ada.head()

Price,Close,High,Low,Open,Volume
Ticker,ADA-USD,ADA-USD,ADA-USD,ADA-USD,ADA-USD
Date,,,,,
2025-01-01,0.917592,0.929422,0.838926,0.843766,926095054
2025-01-02,0.961345,0.979082,0.917597,0.917654,1199916914
2025-01-03,1.090376,1.116099,0.960044,0.961347,2503146669
2025-01-04,1.071400,1.107919,1.058850,1.090352,1175730814
2025-01-05,1.093467,1.107868,1.055279,1.071397,1241263862


**RENDIMIENTO**

Para comprender el atractivo de cada activo, hemos calculado los **rendimientos semanales** de cada criptomoneda. Este indicador nos permite cuantificar la ganancia o pérdida porcentual de una inversión en un período determinado, siendo fundamental para comparar la performance individual de cada activo.

La fórmula empleada es la del cambio porcentual sobre el precio de cierre, lo que nos da una medida precisa de la fluctuación del valor a lo largo del tiempo.

In [152]:
# btc['Close'] tiene los precios de cierre, ahora accedemos correctamente con MultiIndex
btc['retorno'] = btc[('Close', 'BTC-USD')].pct_change()
eth['retorno'] = eth[('Close', 'ETH-USD')].pct_change()
sol['retorno'] = sol[('Close', 'SOL-USD')].pct_change()
ada['retorno'] = ada[('Close', 'ADA-USD')].pct_change()

### Evolución de Rendimientos Acumulados: Gráfico de Líneas

Este gráfico visualiza la trayectoria del rendimiento de una inversión inicial (base 1) para cada una de las criptomonedas (BTC, ETH, SOL, ADA) a lo largo del tiempo. Es una herramienta clave para entender cómo una inversión en cada activo ha crecido (o decrecido) a lo largo del período analizado.

In [205]:
import plotly.express as px

# Calcular los rendimientos acumulados si no se ha hecho aún en esta sesión
# Esto es un respaldo, ya deberían estar calculados en 'rendimientos_acumulados_df'
if 'rendimiento_acumulado' not in btc.columns:
    btc['rendimiento_acumulado'] = (1 + btc['retorno']).cumprod()
    eth['rendimiento_acumulado'] = (1 + eth['retorno']).cumprod()
    sol['rendimiento_acumulado'] = (1 + sol['retorno']).cumprod()
    ada['rendimiento_acumulado'] = (1 + ada['retorno']).cumprod()

    rendimientos_acumulados_df = pd.DataFrame({
        'BTC': btc['rendimiento_acumulado'],
        'ETH': eth['rendimiento_acumulado'],
        'SOL': sol['rendimiento_acumulado'],
        'ADA': ada['rendimiento_acumulado']
    })

# Graficar los rendimientos acumulados
fig_rendimientos_acumulados = px.line(rendimientos_acumulados_df, x=rendimientos_acumulados_df.index, y=rendimientos_acumulados_df.columns,
                                           title='Retorno Acumulado de Criptomonedas (2025-2026)',
                                           labels={'value':'Retorno Acumulado (Base 1)', 'Date':'Fecha', 'variable':'Criptomoneda'},
                                           hover_data={'value':':,.2f'})

fig_rendimientos_acumulados.update_layout(
    xaxis_title='Fecha',
    yaxis_title='Retorno Acumulado (Base 1)',
    hovermode='x unified'
)

fig_rendimientos_acumulados.show()

**¿Qué representa?** Imagina que el 1 de enero de 2025 invertiste 1 unidad monetaria (por ejemplo, 1 dólar o 1 euro) en cada una de estas criptomonedas (BTC, ETH, SOL, ADA). Este gráfico muestra cómo habría evolucionado el valor de esa inversión inicial de 1 unidad a lo largo del tiempo hasta el 26 de mayo de 2026.

**Eje Y** (Retorno Acumulado - Base 1): Los valores en este eje indican el factor de crecimiento. Si el valor es 1.50, significa que tu inversión inicial de 1 unidad ahora vale 1.50 (un retorno del 50%). Si es 0.80, significa que tu inversión de 1 unidad ahora vale 0.80 (una pérdida del 20%).

**Eje X** (Fecha): Muestra la progresión del tiempo, permitiéndote ver el rendimiento diario de esa inversión.

**Líneas Individuales:** Cada línea de color representa una criptomoneda diferente (BTC, ETH, SOL, ADA). Al seguir una línea, puedes observar cómo su valor de inversión acumulado subió o bajó en cualquier día del período.

**Interpretación de Ganancia/Pérdida:** Al final del período, el valor en el eje Y para cada línea te da la ganancia o pérdida total de tu inversión hipotética. Por ejemplo, si la línea de BTC termina en 0.86, significa que tu inversión inicial de 1 ahora vale 0.86, lo que implica una pérdida del 14%.

En resumen, este gráfico es una visualización directa de la ganancia o pérdida total que habrías experimentado si hubieras invertido una cantidad fija en cada una de estas criptomonedas al inicio del período.

### Retorno Acumulado Final por Criptomoneda (Gráfico de Barras)

Para una comparación directa y concisa del desempeño general, este gráfico de barras muestra el retorno acumulado final de cada criptomoneda al cierre del período analizado. Esto facilita la identificación de los activos con mayor crecimiento.

In [206]:
import plotly.express as px

# Asegurarse de que rendimientos_acumulados_df ha sido calculado y tiene la última fila
# Extraer el último valor de retorno acumulado para cada criptomoneda
final_cumulative_returns = rendimientos_acumulados_df.iloc[-1].reset_index()
final_cumulative_returns.columns = ['Criptomoneda', 'Retorno Acumulado Final']

# Crear el gráfico de barras para los retornos acumulados finales, ahora de lado
fig_final_returns_bar = px.bar(final_cumulative_returns, x='Retorno Acumulado Final', y='Criptomoneda', # Intercambiado x e y para barras horizontales
                               title='Retorno Acumulado Final por Criptomoneda (2025-2026)',
                               labels={'Retorno Acumulado Final':'Retorno Acumulado (Base 1)', 'Criptomoneda':'Criptomoneda'},
                               color='Criptomoneda', # Colorea las barras por criptomoneda
                               text='Retorno Acumulado Final',
                               orientation='h') # Asegura la orientación horizontal

fig_final_returns_bar.update_traces(texttemplate='%{text:.2f}', textposition='outside') # Formatear el texto
fig_final_returns_bar.update_layout(uniformtext_minsize=8, uniformtext_mode='hide') # Ajuste de texto

fig_final_returns_bar.show()

En resumen, este gráfico es ideal para mostrar rápidamente a los inversionistas cuál fue el resultado final de cada inversión y facilita la comparación del rendimiento global de las diferentes criptomonedas en el portafolio.

**RIESGO: VOLATILIDAD ANUALIZADA**

En un mercado tan dinámico como el de las criptomonedas, la gestión del riesgo es tan crítica como la búsqueda de rendimientos. Hemos cuantificado este riesgo mediante la **volatilidad anualizada**, una métrica estándar que mide la dispersión de los precios de un activo alrededor de su media. En términos más simples, nos indica cuánto 'se mueve' el precio de una criptomoneda en un año.

Una mayor volatilidad implica un mayor riesgo potencial, pero también la oportunidad de mayores retornos. Nuestro objetivo es equilibrar estos factores inteligentemente. El cálculo se realiza a partir de la desviación estándar de los rendimientos semanales, proyectada a un horizonte anual.

In [194]:
import numpy as np

# Ajustado a 252 días hábiles para volatilidad anualizada
vol_anual_btc = btc['retorno'].std() * np.sqrt(252)
vol_anual_eth = eth['retorno'].std() * np.sqrt(252)
vol_anual_sol = sol['retorno'].std() * np.sqrt(252)
vol_anual_ada = ada['retorno'].std() * np.sqrt(252)

In [208]:
import pandas as pd

# Crear un DataFrame con las volatilidades anualizadas
volatilidades = pd.DataFrame({
    'Criptomoneda': ['BTC', 'ETH', 'SOL', 'ADA'],
    'Volatilidad Anualizada': [vol_anual_btc, vol_anual_eth, vol_anual_sol, vol_anual_ada]
})

# Mostrar el DataFrame de volatilidades
display(volatilidades)

# Crear el gráfico de barras con Plotly Express
fig_volatilidad = px.bar(volatilidades, x='Criptomoneda', y='Volatilidad Anualizada',
                         title='Volatilidad Anualizada de Criptomonedas (2025-2026)',
                         color='Criptomoneda',  # Colorea las barras por criptomoneda
                         labels={'Volatilidad Anualizada': 'Volatilidad Anualizada'},
                         text='Volatilidad Anualizada') # Muestra el valor en la barra
fig_volatilidad.update_traces(texttemplate='%{text:.2f}', textposition='outside') # Formatear el texto
fig_volatilidad.update_layout(uniformtext_minsize=8, uniformtext_mode='hide') # Ajuste de texto
fig_volatilidad.show()

,Criptomoneda,Volatilidad Anualizada
0,BTC,0.376575
1,ETH,0.616964
2,SOL,0.685559
3,ADA,0.863999


### **GRAFICO PRECIOS E HISTOGRAMA DE RETORNOS POR CRIPTOMONEDA**

### Bitcoin (BTC)

In [165]:
# Preparamos el DataFrame `df` a partir de `btc` para que tenga columnas aplanadas y pueda ser usado en Plotly
df_btc = btc.copy()

# Aplanamos las columnas MultiIndex a cadenas de un solo nivel (ej. ('Close', 'BTC-USD') se convierte en 'Close')
df_btc.columns = [col[0] if isinstance(col, tuple) else col for col in df.columns]

In [169]:
import plotly.express as px

# Línea de precios
fig = px.line(df, x=df.index, y='Close', title='Precios Bitcoin 2025-2026')
fig.show()

# Histograma de retornos
fig2 = px.histogram(df, x='retorno', nbins=50, title ='Histograma de Retornos de Bitcoin')
fig2.show()

### Ethereum (ETH)

In [198]:
# Preparamos el DataFrame `df_eth` a partir de `eth` para que tenga columnas aplanadas
df_eth = eth.copy()
df_eth.columns = [col[0] if isinstance(col, tuple) else col for col in df_eth.columns]

# Línea de precios para Ethereum
fig_eth_line = px.line(df_eth, x=df_eth.index, y='Close', title='Precios Ethereum 2025-2026')
fig_eth_line.show()

# Histograma de retornos para Ethereum
fig_eth_hist = px.histogram(df_eth, x='retorno', nbins=50, title='Histograma de Retornos de Ethereum')
fig_eth_hist.show()

### Solana (SOL)

In [170]:
# Preparamos el DataFrame `df_sol` a partir de `sol` para que tenga columnas aplanadas
df_sol = sol.copy()
df_sol.columns = [col[0] if isinstance(col, tuple) else col for col in df_sol.columns]

# Línea de precios para Solana
fig_sol_line = px.line(df_sol, x=df_sol.index, y='Close', title='Solana 2025-2026')
fig_sol_line.show()

# Histograma de retornos para Solana
fig_sol_hist = px.histogram(df_sol, x='retorno', nbins=50, title='Histograma de Retornos de Solana')
fig_sol_hist.show()

### Cardano (ADA)

In [171]:
# Preparamos el DataFrame `df_ada` a partir de `ada` para que tenga columnas aplanadas
df_ada = ada.copy()
df_ada.columns = [col[0] if isinstance(col, tuple) else col for col in df_ada.columns]

# Línea de precios para Cardano
fig_ada_line = px.line(df_ada, x=df_ada.index, y='Close', title='Cardano 2025-2026')
fig_ada_line.show()

# Histograma de retornos para Cardano
fig_ada_hist = px.histogram(df_ada, x='retorno', nbins=50, title='Histograma de Retornos de Cardano')
fig_ada_hist.show()

### Evolución Individual de Precios: Gráfico de Líneas Múltiples

Para una comparación directa y clara de la trayectoria de precios de cada criptomoneda, presentamos un gráfico de líneas donde Bitcoin, Ethereum, Solana y Cardano tienen sus propias líneas. Esto permite observar las tendencias individuales y cómo se mueven en relación entre sí.

In [209]:
# Crear el gráfico de líneas con las criptomonedas individuales
fig_individual_lines = px.line(combined_prices.drop(columns=['Total Combined Price'], errors='ignore'),
                               x=combined_prices.index, y=combined_prices.columns[:-1] if 'Total Combined Price' in combined_prices.columns else combined_prices.columns,
                               title='Evolución Individual de Precios de Criptomonedas (2025-2026)',
                               labels={'value':'Precio (USD)', 'Date':'Fecha', 'variable':'Criptomoneda'},
                               hover_data={'value':':,.2f'})

fig_individual_lines.update_layout(
    xaxis_title='Fecha',
    yaxis_title='Precio (USD)',
    hovermode='x unified'
)

fig_individual_lines.show()

### 4. Interpretación Estratégica: ¿Dónde Poner su Capital?

La tabla de volatilidades anualizadas nos ofrece información clave para nuestra estrategia de inversión:

*   **Bitcoin (BTC):** Con la menor volatilidad entre las opciones analizadas, BTC se posiciona como el ancla de nuestro fondo. Su relativa estabilidad lo convierte en una opción robusta para la preservación de capital dentro del ecosistema cripto.
*   **Ethereum (ETH):** Presenta una volatilidad moderada. Nos permite capturar crecimiento significativo sin asumir el riesgo extremo de activos más especulativos, ofreciendo un balance atractivo.
*   **Solana (SOL):** Muestra una volatilidad alta, lo que implica un mayor riesgo, pero también un potencial de crecimiento superior. Incluimos SOL con una asignación estratégica y controlada para aprovechar sus oportunidades de alto rendimiento.
*   **Cardano (ADA):** Exhibe la mayor volatilidad de todas las criptomonedas analizadas, lo que implica el riesgo más alto, pero también el mayor potencial de crecimiento. Incluimos ADA con una asignación muy estratégica y controlada para buscar retornos superiores, contribuyendo a la diversificación y al potencial alcista del fondo.

Nuestra estrategia de inversión busca un balance óptimo entre la seguridad (BTC) y el potencial de crecimiento (ETH, SOL, ADA), construyendo una cartera diversificada que se alinee con los objetivos de su fondo de pensión.

### 5. Conclusión: Invierta con Inteligencia y Confianza

Nuestro enfoque se basa en un análisis riguroso de datos históricos para entender profundamente el riesgo y el rendimiento de cada activo. Al invertir con nuestro fondo, usted elige una gestión activa y basada en evidencia que busca maximizar sus retornos ajustados al riesgo.

Estamos convencidos de que esta aproximación metódica es clave para navegar con éxito el mercado de criptomonedas y para hacer crecer su capital de manera sostenida.

**¡Estamos listos para discutir cómo nuestro fondo puede ser el socio estratégico ideal para sus metas de inversión!**

### **Interpretando la Dispersion absoluta de precios con .describe()**


Aunque la volatilidad anualizada es ideal para comparar activos entre sí, la desviación estándar ('std') que nos proporciona el método `.describe()` para el precio de cierre de cada criptomoneda nos da una idea de la dispersión de sus precios durante el período de 2025-2026.

In [197]:
btc.describe()

Price,Close,High,Low,Open,Volume,retorno,rendimiento_acumulado
Ticker,BTC-USD,BTC-USD,BTC-USD,BTC-USD,BTC-USD,,
count,490.000000,490.000000,490.000000,490.000000,4.900000e+02,489.000000,489.000000
mean,95089.170161,96623.977121,93458.259319,95112.125263,5.007445e+10,-0.000013,1.007104
std,15829.784780,15791.183460,15827.352102,15818.365639,2.296137e+10,0.023722,0.167825
min,62702.097656,64992.156250,60074.203125,62704.453125,9.863214e+09,-0.141299,0.664078
25%,84138.291016,85433.478516,82511.548828,84136.009766,3.555885e+10,-0.012608,0.891007
50%,95532.332031,96711.921875,94050.968750,95528.843750,4.735649e+10,-0.000024,1.011860
75%,108404.519531,109746.894531,107146.404297,108402.914062,6.301318e+10,0.011793,1.148180
max,124752.531250,126198.070312,123196.046875,124752.140625,1.817464e+11,0.125248,1.321255


In [156]:
eth.describe()

Price,Close,High,Low,Open,Volume,retorno
Ticker,ETH-USD,ETH-USD,ETH-USD,ETH-USD,ETH-USD,
count,490.000000,490.000000,490.000000,490.000000,4.900000e+02,489.000000
mean,2887.825277,2964.949279,2808.291627,2889.736449,2.678262e+10,0.000054
std,839.082374,855.686008,822.686980,839.003131,1.408574e+10,0.038865
min,1472.553101,1587.538574,1386.799316,1472.601440,6.117037e+09,-0.150138
25%,2189.441895,2254.970337,2104.536499,2189.618774,1.677306e+10,-0.021951
50%,2741.793091,2824.604126,2655.836670,2753.526978,2.350284e+10,0.000180
75%,3398.467285,3493.351196,3303.020020,3398.260071,3.437880e+10,0.017759
max,4831.348633,4953.732910,4714.472656,4831.088379,9.773662e+10,0.217988


In [196]:
sol.describe()

Price,Close,High,Low,Open,Volume,retorno,rendimiento_acumulado
Ticker,SOL-USD,SOL-USD,SOL-USD,SOL-USD,SOL-USD,,
count,490.000000,490.000000,490.000000,490.000000,4.900000e+02,489.000000,489.000000
mean,151.359542,156.141062,146.812535,151.567653,5.288201e+09,-0.000712,0.780264
std,45.222327,46.945895,43.551584,45.158895,3.157827e+09,0.043186,0.233284
min,77.753319,79.578949,68.691818,77.754417,1.442831e+09,-0.204344,0.401051
25%,124.753428,127.748287,122.286053,124.978304,3.450653e+09,-0.026230,0.643172
50%,146.861343,150.126457,143.005882,147.000824,4.460560e+09,-0.000827,0.757173
75%,186.220352,190.846645,179.911331,186.679764,6.341753e+09,0.020656,0.958053
max,261.869751,294.334961,248.751602,261.872437,3.317296e+10,0.242350,1.350723


In [148]:
ada.describe()

Price,Close,High,Low,Open,Volume,retorno
Ticker,ADA-USD,ADA-USD,ADA-USD,ADA-USD,ADA-USD,
count,490.000000,490.000000,490.000000,490.000000,4.900000e+02,0.0
mean,0.603738,0.624771,0.583839,0.604925,1.013031e+09,NaN
std,0.236374,0.246509,0.226911,0.236114,8.067333e+08,NaN
min,0.236272,0.246875,0.226238,0.236272,2.094067e+08,NaN
25%,0.375927,0.391137,0.363803,0.378901,5.565741e+08,NaN
50%,0.648166,0.670667,0.627578,0.651306,7.850031e+08,NaN
75%,0.791109,0.815858,0.765431,0.791937,1.199072e+09,NaN
max,1.135122,1.163675,1.088220,1.135109,7.647922e+09,NaN


Observando el `std` de la columna `Close` para cada criptomoneda:

*   **BTC:** La desviación estándar de su precio de cierre es de aproximadamente **15829.78 USD**. Esto indica cuánto se desvía el precio de Bitcoin de su media durante 2025-2026.
*   **ETH:** Su `std` es de aproximadamente **839.08 USD**. Comparado con BTC, su variación absoluta en precio es menor, lo cual es esperable dado su precio unitario más bajo.
*   **SOL:** Con un `std` de aproximadamente **45.22 USD**. Aunque en valor absoluto es el más bajo, en términos porcentuales respecto a su precio medio, esto aún representa una variación significativa.
*   **ADA:** Su `std` es de aproximadamente **0.24 USD**. Al igual que SOL, en términos de valor absoluto, es la más baja, pero su volatilidad relativa sigue siendo notable.

**Conclusión con `.describe()`:**

Si bien los valores absolutos de `std` varían debido a los diferentes rangos de precios de cada criptomoneda, la desviación estándar nos confirma que los precios de **BTC** tuvieron la mayor dispersión en términos monetarios absolutos, mientras que SOL tuvo una alta dispersión relativa.

Para una comparación más justa de riesgo relativo, la **volatilidad anualizada** (que ya calculamos) es la métrica preferida, donde **ADA** se destacó como la más volátil.

### Máxima Caída (Maximum Drawdown - MDD)

In [210]:
import numpy as np
import pandas as pd

def calculate_mdd(cumulative_returns):
    if cumulative_returns.empty or cumulative_returns.isnull().all():
        return np.nan
    # Asegurarse de que no haya NaN en el cálculo del MDD si se pueden omitir los primeros
    temp_returns = cumulative_returns.dropna()
    if temp_returns.empty:
        return np.nan

    # Calcula el máximo histórico (peak) hasta la fecha actual
    running_max = temp_returns.cummax()
    # Calcula la caída actual desde el máximo histórico
    drawdown = (temp_returns / running_max) - 1
    # El Maximum Drawdown es la caída más grande
    mdd = drawdown.min()
    return mdd

# Calcular el MDD para cada criptomoneda
mdd_btc = calculate_mdd(btc['rendimiento_acumulado'])
mdd_eth = calculate_mdd(eth['rendimiento_acumulado'])
mdd_sol = calculate_mdd(sol['rendimiento_acumulado'])
mdd_ada = calculate_mdd(ada['rendimiento_acumulado'])

# Crear un DataFrame para mostrar los resultados
mdd_df = pd.DataFrame({
    'Criptomoneda': ['BTC', 'ETH', 'SOL', 'ADA'],
    'Maximum Drawdown': [mdd_btc, mdd_eth, mdd_sol, mdd_ada]
})

# Mostrar el DataFrame
display(mdd_df)

# Crear un gráfico de barras para el MDD
fig_mdd = px.bar(mdd_df, x='Criptomoneda', y='Maximum Drawdown',
                 title='Maximum Drawdown (Peor Caída Histórica) por Criptomoneda',
                 labels={'Maximum Drawdown':'MDD', 'Criptomoneda':'Criptomoneda'},
                 color='Criptomoneda', # Colorea las barras por criptomoneda
                 text='Maximum Drawdown') # Muestra el valor en la barra
fig_mdd.update_traces(texttemplate='%{text:.2%}', textposition='outside') # Formatear el texto como porcentaje
fig_mdd.update_layout(uniformtext_minsize=8, uniformtext_mode='hide') # Ajuste de texto
fig_mdd.show()

,Criptomoneda,Maximum Drawdown
0,BTC,-0.497388
1,ETH,-0.622945
2,SOL,-0.703084
3,ADA,-0.791853


Este gráfico de barras muestra la *Máxima Caída (Maximum Drawdown - MDD)* para cada criptomoneda. El MDD es una métrica clave de riesgo que representa la mayor pérdida porcentual desde un pico hasta un valle subsiguiente antes de alcanzar un nuevo pico. En otras palabras, mide la peor pérdida que un inversor habría sufrido si hubiera comprado en el pico y vendido en el valle más bajo después de ese pico.

**Interpretación:**

*   Un MDD de -0.50 (-50%) significa que, en algún momento, la criptomoneda perdió la mitad de su valor desde su máximo histórico hasta ese punto.
*   Cuanto mayor (menos negativo) sea el MDD, menor fue la peor caída, lo que indica una mayor resiliencia del activo.

Esta visualización es fundamental para entender el riesgo histórico de una inversión, mostrando el peor escenario de pérdida que se ha presentado en el periodo analizado.

### Porcentaje de Días con Rendimiento Negativo (Frecuencia de Pérdidas)

Esta métrica nos muestra la proporción de días en los que el rendimiento de una criptomoneda fue negativo, lo que nos da una idea directa de la frecuencia de las pérdidas. Es un indicador clave para evaluar el riesgo asociado a la volatilidad a la baja.

In [204]:
import pandas as pd
import plotly.express as px

# Calcular el porcentaje de días con rendimiento negativo para cada criptomoneda
neg_ret_btc = (btc['retorno'] < 0).sum() / btc['retorno'].dropna().count()
neg_ret_eth = (eth['retorno'] < 0).sum() / eth['retorno'].dropna().count()
neg_ret_sol = (sol['retorno'] < 0).sum() / sol['retorno'].dropna().count()
neg_ret_ada = (ada['retorno'] < 0).sum() / ada['retorno'].dropna().count()

# Crear un DataFrame para mostrar los resultados
neg_returns_df = pd.DataFrame({
    'Criptomoneda': ['BTC', 'ETH', 'SOL', 'ADA'],
    'Porcentaje Días Negativos': [neg_ret_btc, neg_ret_eth, neg_ret_sol, neg_ret_ada]
})

# Mostrar el DataFrame
display(neg_returns_df)

# Crear un gráfico de barras para el porcentaje de días negativos
fig_neg_returns = px.bar(neg_returns_df, x='Criptomoneda', y='Porcentaje Días Negativos',
                         title='Porcentaje de Días con Rendimiento Negativo (2025-2026)',
                         labels={'Porcentaje Días Negativos':'% Días Negativos', 'Criptomoneda':'Criptomoneda'},
                         color='Criptomoneda', # Colorea las barras por criptomoneda
                         text='Porcentaje Días Negativos') # Muestra el valor en la barra
fig_neg_returns.update_traces(texttemplate='%{text:.2%}', textposition='outside') # Formatear el texto como porcentaje
fig_neg_returns.update_layout(uniformtext_minsize=8, uniformtext_mode='hide') # Ajuste de texto
fig_neg_returns.show()

,Criptomoneda,Porcentaje Días Negativos
0,BTC,0.503067
1,ETH,0.496933
2,SOL,0.515337
3,ADA,0.543967


Este gráfico de barras muestra claramente la **frecuencia con la que cada criptomoneda ha tenido un rendimiento diario negativo**. Una barra más alta indica que el activo ha registrado pérdidas con mayor asiduidad durante el periodo analizado.

**Interpretación para inversionistas:**

*   Un alto porcentaje de días negativos sugiere una mayor inestabilidad y un riesgo más frecuente de ver el valor de la inversión disminuir en el corto plazo. Esto es importante para aquellos inversionistas con menor tolerancia al riesgo o que planean realizar operaciones frecuentes.
*   Un bajo porcentaje de días negativos, por el contrario, indica una mayor consistencia en el rendimiento positivo, lo que puede ser percibido como una señal de mayor estabilidad, aunque no necesariamente de mayor rentabilidad absoluta.

Esta métrica complementa la volatilidad y el Maximum Drawdown, ofreciendo una perspectiva sobre la **constancia del riesgo de pérdida**, en lugar de solo la magnitud de las caídas.